### PR TITLE
Filter cases by OMIM terms in cases and dashboard pages

### LINKED ISSUE #3221: Search and filter cases by diagnoses (OMIM)
Looks like cases are NOT searchable by OMIM terms:

image

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3221

- Add filtering of cases by OMIM terms in both cases and dashboard pages
- Introduce placeholder, to give case query example in dashboard page 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. Install on cg-vm1 and filter using one of more OMIM terms
2. Filter also by OMIM terms with blank field. It should return cases without diagnoses

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/server/blueprints/institutes/static/form_scripts.js [hunk 1]
```diff
@@ -1,26 +1,3 @@
-var searchTerm = document.getElementById('search_term');
```
Comment (northwestwitch) [inline scout/server/blueprints/institutes/static/form_scripts.js:1]: Moved to constants/cases/case_tags, so it ca be used by both cases and dashboard pages

**Comment by northwestwitch** [comment]
This is ready for review now!

File: scout/adapter/mongo/case.py [hunk 2]
```diff
@@ -177,6 +177,25 @@ def _populate_name_query(self, query, name_query, owner=None, collaborator=None)
...
+            if query_term != "":
+                # User might have provided multiple query terms
+                omim_terms = [term for term in query_term.replace(" ", "").split(",")]
+                query["$or"] = [
+                    {
+                        "diagnosis_phenotypes.disease_id": {"$in": omim_terms}
+                    },  # OMIM phenotypes are assigned as a list of dictionaries
+                    {
+                        "diagnosis_phenotypes": {"$in": omim_terms}
+                    },  # old way of saving OMIM terms --> list of ids [616538,611277]
```
Comment (dnil) [inline scout/adapter/mongo/case.py:190]: 🏅

File: scout/adapter/mongo/case.py [hunk 3]
```diff
@@ -177,6 +177,25 @@ def _populate_name_query(self, query, name_query, owner=None, collaborator=None)
...
+                ]
+
+            else:  # query for cases with no HPO terms
```
Comment (dnil) [inline scout/adapter/mongo/case.py:193]:
```suggestion
            else:  # query for cases with no OMIM diagnosis
```

File: scout/server/blueprints/institutes/static/form_scripts.js [hunk 4]
```diff
@@ -1,26 +1,3 @@
...
-var sel = document.getElementById('search_type');
```
Comment (dnil) [inline scout/server/blueprints/institutes/static/form_scripts.js:2]: 👍

File: scout/server/blueprints/dashboard/templates/dashboard/dashboard_general.html [hunk 5]
```diff
@@ -159,29 +159,41 @@ <h1>Pedigree info</h1>
...
+        searchTerm.disabled = (sel.value == "");
+        searchTerm.placeholder="Search term";
+      }
+      // Reset search
+      searchTerm.value="";
+    };
 
     $(document).ready(function(){
-      text.disabled = (sel.value == "");
+      searchTerm.disabled = (sel.value == "");
```
Comment (dnil) [inline scout/server/blueprints/dashboard/templates/dashboard/dashboard_general.html:189]: 👍

File: scout/constants/case_tags.py [hunk 6]
```diff
@@ -136,32 +136,46 @@
 SAMPLE_SOURCE = dict((i, el) for i, el in enumerate(SOURCES))
 
 CASE_SEARCH_TERMS = {
-    "case": {"label": "Case or Individual Name", "prefix": "case:"},
+    "case": {"label": "Case or Individual Name", "prefix": "case:", "placeholder": "example:18201"},
```
Comment (dnil) [inline scout/constants/case_tags.py:139]: 👍

File: scout/server/blueprints/dashboard/templates/dashboard/dashboard_general.html [hunk 7]
```diff
@@ -159,29 +159,41 @@ <h1>Pedigree info</h1>
...
-    });
+    if(document.getElementById('download_verified')){
+      document.querySelector("#download_verified").addEventListener("submit", function(e){
+        // Avoid page spinner being stuck on download verified variants
+        $(window).unbind('beforeunload');
+      });
+    }
 
-    var sel = document.getElementById("search_type");
-    var text = document.getElementById("search_term");
```
Comment (dnil) [inline scout/server/blueprints/dashboard/templates/dashboard/dashboard_general.html:168]: Ah, that one is still used down below - like line 197 in hide_div - search and replace! 😁

**Review by dnil — APPROVED** [review]
Works like a charm, and super nice with the generalised hints for dashboard!

The only thing was a js warning on the dashboard page for the changed name for search terms!
Screenshot 2022-03-08 at 14 38 36

File: scout/server/blueprints/dashboard/templates/dashboard/dashboard_general.html (hunk 7 above)
Comment (northwestwitch) [reply scout/server/blueprints/dashboard/templates/dashboard/dashboard_general.html:168]: Fixed!

image


image

**Pull request merged**